h/resources.py

```diff
@@ -106,6 +108,32 @@ def __getitem__(self, client_id):
             raise KeyError()
 
 
+class OrganizationFactory(object):
+    def __init__(self, request):
+        self.request = request
+
+    def __getitem__(self, pubid):
+        try:
+            return self.request.db.query(Organization).filter_by(pubid=pubid).one()
+        except exc.NoResultFound:
+            raise KeyError()
+
+
+class OrganizationLogoFactory(object):
+    def __init__(self, request):
+        self.request = request
+        self.organization_factory = OrganizationFactory(self.request)
+
+    def __getitem__(self, pubid):
+        # This will raise KeyError if the organization doesn't exist.
+        organization = self.organization_factory[pubid]
+
+        if not organization.logo:
+            raise KeyError()
```

Comment:
And _this_ `KeyError` causes a 404 if the group does exist but has no logo

Reply:
OK, so in any sort of API response, one would need to check the organization directly make sure it has a logo and not return a URL if not? Wouldn't want to return logo URLs in organization objects via API if the logo doesn't actual exist, right?